kb/deadline-exceeded-troubleshooting.md

```diff
@@ -0,0 +1,116 @@
+---
+slug: deadline-exceeded-troubleshooting
+title: DeadlineExceeded error troubleshooting
+tags:
+  - kb-article
+date: 2022-11-09T00:00:00Z
+---
+
+All client-side requests made to the [Temporal Cluster](/concepts/what-is-a-temporal-cluster) are [gRPC requests](https://grpc.io/docs/what-is-grpc/core-concepts/#deadlines).
+Sometimes, when these requests can't be completed, you'll see this particular error message: `Context: deadline exceeded`.
+Network interruptions, timeouts, server overload, and Query errors are some of the reasons that can cause this error.
+
+The following sections discuss the nature of this error and how to troubleshoot it.
+
+#### Check Frontend Service logs
+
+:::note
+
+Cloud users cannot access some of the logs needed to diagnose the source of the error.
+
+If you're using Temporal Cloud, check your Workflow History for error messages and create a [support ticket](https://support.temporal.io/) with Temporal.
+Provide the full error message in your ticket.
+
+:::
+
+[Frontend Service](/clusters/#frontend-service) logs can show which parts of the Cluster aren't working.
+
+Verify that the Frontend Service is connected by opening the Web UI in your browser.
+
+Alternatively, OSS users can check that Frontend and other service are running with `tctl`.
+
+```
+
+tctl --address frontendAddress:frontendPort cluster health
+
+```
+
+Use `grpc-health-probe` to check the [Frontend Service](/clusters#frontend-service), [Matching Service](/clusters#matching-service), and [History Service](/clusters#history-service).
+
+```
+
+./grpc-health-probe -addr=frontendAddress:frontendPort -service=temporal.api.workflowservice.v1.WorkflowService
+
+./grpc-health-probe -addr=matchingAddress:matchingPort -service=temporal.api.workflowservice.v1.MatchingService
+
+./grpc-health-probe -addr=historyAddress:historyPort -service=temporal.api.workflowservice.v1.HistoryService
+
+```
+
+Logs can also be used to find Client [Query](/workflows#queries) requests that failed with this error.
+
+#### Check your Cluster metrics
+
+Cluster metrics can be used to detect issues (such as 'resource exhausted') that implact Cluster health.
```

Comment:
```suggestion
Cluster metrics can be used to detect issues (such as `resource exhausted`) that impact Cluster health.
```
I can't keep "implact" here, but I'm gonna find a way to work it into conversations every chance I get. (Definition: A mashup of "impact" and "splat.")